src/dataflow/src/decode/mod.rs

```diff
@@ -393,6 +399,74 @@ where
     })
 }
 
+fn decode_cdcv2<G: Scope<Timestamp = Timestamp>>(
+    stream: &Stream<G, SourceOutput<Vec<u8>, Vec<u8>>>,
+    schema: &str,
+    registry: Option<ccsr::ClientConfig>,
+) -> (Stream<G, (Row, Timestamp, Diff)>, Option<Box<dyn Any>>) {
+    let mut resolver = ConfluentAvroResolver::new(schema, registry).unwrap(); // We will have already checked validity of the schema by now, so this can't fail.
+    let channel = Rc::new(RefCell::new(VecDeque::new()));
+    let activator: Rc<RefCell<Option<SyncActivator>>> = Rc::new(RefCell::new(None));
+    let mut vector = Vec::new();
+    let stream = stream.unary::<(), _, _, _>(
+        SourceOutput::<Vec<u8>, Vec<u8>>::position_value_contract(),
+        "CDCv2-Decode",
+        {
+            let channel = channel.clone();
+            let activator = activator.clone();
+            move |_, _| {
+                move |input, _output| {
+                    input.for_each(|_time, data| {
+                        data.swap(&mut vector);
+                        for data in vector.drain(..) {
+                            let (mut data, schema) = match block_on(resolver.resolve(&data.value)) {
+                                Ok(ok) => ok,
+                                Err(e) => {
+                                    error!("Failed to get schema info for CDCv2 record: {}", e);
+                                    continue;
+                                }
+                            };
+                            let d = GeneralDeserializer {
+                                schema: schema.top_node(),
+                            };
+                            let dec = interchange::avro::cdc_v2::Decoder;
+                            let message = match d.deserialize(&mut data, dec) {
+                                Ok(ok) => ok,
+                                Err(e) => {
+                                    error!("Failed to deserialize avro message: {}", e);
+                                    continue;
+                                }
+                            };
+                            channel.borrow_mut().push_back(message);
+                        }
+                    });
+                    if let Some(activator) = activator.borrow_mut().as_mut() {
+                        activator.activate().unwrap()
+                    }
+                }
+            }
+        },
+    );
+    struct VdIterator<T>(Rc<RefCell<VecDeque<T>>>);
+    impl<T> Iterator for VdIterator<T> {
+        type Item = T;
+        fn next(&mut self) -> Option<T> {
+            // XXX yield after 10ms ?
```

Comment:
Is the status of this "ideally the Kafka source will yield often enough that we will get a break too"? I could imagine that being fine, but if you have a better take on what we should do, good to know!

Reply:
I'll wrap this in YieldingIter from dd, just to be sure.